packages/model-viewer/src/test/features/scene-graph/model-spec.ts

```diff
@@ -74,5 +79,58 @@ suite('scene-graph/model', () => {
 
       expect(collectedMaterials.size).to.be.equal(materials.size);
     });
+
+    suite('Model Variants', () => {
+      test('Switch variant and lazy load', async () => {
+        const threeGLTF = await loadThreeGLTF(SHEEN_CHAIR_GLB_PATH);
+        const model = new Model(CorrelatedSceneGraph.from(threeGLTF));
+
+        expect(model[$materials][4][$correlatedObjects]).to.be.null;
+        expect(model[$materials][4][$lazyLoadGLTFInfo]).to.not.be.undefined;
+
+        expect(model[$materials][5][$correlatedObjects]).to.be.null;
+        expect(model[$materials][4][$lazyLoadGLTFInfo]).to.not.be.undefined;
+
+        await model[$switchVariant]('Peacock Velvet');
+
+        expect(model[$materials][4][$correlatedObjects]).to.not.be.null;
+        expect(model[$materials][4][$lazyLoadGLTFInfo]).to.be.undefined;
```

Comment:
This is because you've loaded all the variants, right?

Reply:
Correct, when the variant loads the LazyLoader reference is removed.